src/NuGet.Core/NuGet.PackageManagement/Utility/PackagesConfigLockFileUtility.cs

```diff
@@ -80,6 +83,94 @@ public static string GetPackagesLockFilePath(string projectPath, string nuGetLoc
             return PackagesLockFileUtilities.GetNuGetLockFilePath(projectPath, projectName);
         }
 
+        public static IReadOnlyList<IRestoreLogMessage> ValidatePackagesConfigLockFiles(
+            string projectFile,
+            string packagesConfigFile,
+            string projectName,
+            string nuGetLockFilePath,
+            string restorePackagesWithLockFile,
+            NuGetFramework projectTfm,
+            string packagesFolderPath,
+            bool lockedMode,
+            CancellationToken token)
+        {
+            var lockFilePath = GetPackagesLockFilePath(Path.GetDirectoryName(packagesConfigFile), nuGetLockFilePath, projectName);
+            var lockFileExists = File.Exists(lockFilePath);
+            var lockFileOptIn = MSBuildStringUtility.GetBooleanOrNull(restorePackagesWithLockFile);
```

Comment:
that will change behaviour since the boolean is nullable. The line below does `== true` which will be false if null, and the following if statement checks `== false` which is also false if null. If this declaration does the equality comparison, I lose the ability to be false both times it's currently used.